src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -52,10 +52,10 @@ public sealed class RSessionProvider : IRSessionProvider {
         public event EventHandler<BrokerStateChangedEventArgs> BrokerStateChanged;
         public event EventHandler<HostLoadChangedEventArgs> HostLoadChanged;
 
-        public RSessionProvider(ICoreServices services, IConsole callback = null) {
+        public RSessionProvider(ICoreShell coreShell, IConsole callback = null) {
```

Comment:
We can try that too. Changes can go step by step. Need to be able to get rid of MEF in the editor so I can start splitting it.